### PR TITLE
Update UI of the small Link panel

### PR DESCRIPTION
**Pull request by filipsobol**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Feature (core): Add `settings` icon. See #17230.

Feature (link): Update UI of the small Link panel. See #17230.

### CODE REVIEW

File: packages/ckeditor5-ui/src/formheader/formheaderview.ts [hunk 1]
```diff
@@ -52,6 +52,11 @@ export default class FormHeaderView extends View {
 	 */
 	public readonly iconView?: IconView;
 
+	/**
+	 * Form header options passed to the constructor.
+	 */
+	private options: FormHeaderViewOptions;
```
Comment (niegowski) [inline packages/ckeditor5-ui/src/formheader/formheaderview.ts:58]:
```suggestion
	private _options: FormHeaderViewOptions;
```

File: packages/ckeditor5-ui/src/formheader/formheaderview.ts [hunk 2]
```diff
@@ -61,41 +66,69 @@ export default class FormHeaderView extends View {
...
-			icon?: string | null;
-		} = {}
+		options: FormHeaderViewOptions = {}
 	) {
 		super( locale );
 
 		const bind = this.bindTemplate;
 
+		this.options = options;
+		this.iconView = this._createIcon();
```
Comment (niegowski) [inline packages/ckeditor5-ui/src/formheader/formheaderview.ts:76]: Is this always required? Maybe we could create it only when it is needed?

File: packages/ckeditor5-ui/src/formheader/formheaderview.ts [hunk 3]
```diff
@@ -61,41 +66,69 @@ export default class FormHeaderView extends View {
...
+
 		this.set( 'label', options.label || '' );
 		this.set( 'class', options.class || null );
 
 		this.children = this.createCollection();
 
+		if ( this.options.left ) {
+			this.children.add( this.options.left );
+		}
```
Comment (niegowski) [inline packages/ckeditor5-ui/src/formheader/formheaderview.ts:85]: What does `left` mean? Maybe we could name it more precisely? Same for `right`

File: packages/ckeditor5-ui/src/formheader/formheaderview.ts [hunk 4]
```diff
@@ -61,41 +66,69 @@ export default class FormHeaderView extends View {
...
 			attributes: {
 				class: [
 					'ck',
 					'ck-form__header',
 					bind.to( 'class' )
-				]
+				],
+
+				// https://github.com/ckeditor/ckeditor5-link/issues/90
+				tabindex: '-1'
```
Comment (niegowski) [inline packages/ckeditor5-ui/src/formheader/formheaderview.ts:107]: Why is this needed? Does the form header should be focusable on its own?

**Review by niegowski — COMMENTED** [review]
Partial review.

File: packages/ckeditor5-ui/src/formheader/formheaderview.ts (hunk 4 above)
Comment (filipsobol) [reply packages/ckeditor5-ui/src/formheader/formheaderview.ts:107]: Here and in a few other places we use this workaround for an issue where `Escape` stops working, when a given element (header in this case) in clicked on.

File: packages/ckeditor5-ui/src/formheader/formheaderview.ts (hunk 3 above)
Comment (filipsobol) [reply packages/ckeditor5-ui/src/formheader/formheaderview.ts]: Renamed to `leftSlot` and `rightSlot`.

File: packages/ckeditor5-ui/src/formheader/formheaderview.ts (hunk 2 above)
Comment (filipsobol) [reply packages/ckeditor5-ui/src/formheader/formheaderview.ts:76]: In the function itself we check if an icon was passed, and if not, then it returns `null`.

**Comment by niegowski** [comment]
I just noticed that you did not used an epic branch as a base for this PR, it was supposed to be `ck/epic/17230-linking-experience`

File: packages/ckeditor5-link/src/linkui.ts [hunk 5]
```diff
@@ -273,6 +281,21 @@ export default class LinkUI extends Plugin {
...
+	/**
+	 * Creates a bookmarks button view.
+	 */
+	private _createBookmarksButton(): LinkButtonView {
+		const locale = this.editor.locale!;
+		const t = locale.t;
+		const bookmarksButton = new LinkButtonView( locale );
+
+		bookmarksButton.set( {
+			label: t( 'Bookmarks' )
```
Comment (pszczesniak) [inline packages/ckeditor5-link/src/linkui.ts:293]: Missing in `contexts.json`

File: packages/ckeditor5-link/src/ui/linkformview.ts [hunk 6]
```diff
@@ -242,49 +282,146 @@ export default class LinkFormView extends View {
...
+
+	/**
+	 * Creates a save button view that inserts the link.
+	 */
+	private _createSaveButton(): ButtonView {
+		const t = this.locale!.t;
+		const saveButton = new ButtonView( this.locale );
+
+		saveButton.set( {
+			label: t( 'Insert' ),
```
Comment (pszczesniak) [inline packages/ckeditor5-link/src/ui/linkformview.ts:326]: To remember - probably we will need to change the label when link already exists (editing). @dwandeljanicka am i right?

File: packages/ckeditor5-link/src/ui/linkformview.ts [hunk 7]
```diff
@@ -242,49 +282,146 @@ export default class LinkFormView extends View {
...
+	}
+
+	/**
+	 * Creates a labeled input view for the "Displayed text" field.
+	 */
+	private _createDisplayedTextInput(): LabeledFieldView<InputTextView> {
+		const t = this.locale!.t;
+		const labeledInput = new LabeledFieldView( this.locale, createLabeledInputText );
+
+		labeledInput.label = t( 'Displayed text' );
```
[the remark this answers is not in the document]
Comment (filipsobol) [reply packages/ckeditor5-link/src/ui/linkformview.ts:407]: Fixed in https://github.com/ckeditor/ckeditor5/pull/17265/commits/9eae802ffd930116379dd9eaeb87323210912edd.

File: packages/ckeditor5-link/src/ui/linkformview.ts [hunk 8]
```diff
@@ -74,10 +87,20 @@ export default class LinkFormView extends View {
 	private readonly _manualDecoratorSwitches: ViewCollection<SwitchButtonView>;
 
 	/**
-	 * A collection of child views in the form.
+	 * A collection of children elements.
```
Comment (niegowski) [inline packages/ckeditor5-link/src/ui/linkformview.ts:90]:
```suggestion
	 * A collection of child views.
```

File: packages/ckeditor5-link/src/ui/linkformview.ts [hunk 9]
```diff
@@ -242,49 +282,146 @@ export default class LinkFormView extends View {
...
+		return saveButton;
+	}
+
+	/**
+	 * Creates a header view for the form.
+	 */
+	private _createHeaderView(): FormHeaderView {
+		const t = this.locale!.t;
+
+		const form = new FormHeaderView( this.locale, {
```
Comment (niegowski) [inline packages/ckeditor5-link/src/ui/linkformview.ts:342]:
```suggestion
		const header = new FormHeaderView( this.locale, {
```

File: packages/ckeditor5-link/theme/linkform.css [hunk 10]
```diff
@@ -5,42 +5,81 @@
...
+	/* Form */
+	& .ck-responsive-form.ck-link__form {
+		display: flex;
+		flex-direction: column;
+
+		& > * {
+			margin: 0;
+
+			@mixin ck-media-phone {
+				margin: var(--ck-spacing-medium);
```
Comment (pszczesniak) [inline packages/ckeditor5-link/theme/linkform.css:31]:
```suggestion
				margin: var(--ck-spacing-large) var(--ck-spacing-medium)  var(--ck-spacing-medium)  var(--ck-spacing-medium);
```

File: packages/ckeditor5-link/theme/linkform.css [hunk 11]
```diff
@@ -5,42 +5,81 @@
...
+				flex-direction: column;
+
+				&>*,
+				& .ck-button-insert {
+					margin: 0;
+				}
+
+				& .ck-button-insert {
+					margin-top: var(--ck-spacing-large);
+				}
```
Comment (pszczesniak) [inline packages/ckeditor5-link/theme/linkform.css:64]:
```suggestion
				& > .ck {
					margin: 0;
				}

				& .ck-button-insert {
					margin-top: var(--ck-spacing-large);
					justify-content: center;
				}
```

File: packages/ckeditor5-link/theme/linkform.css [hunk 12]
```diff
@@ -5,42 +5,81 @@
...
+		margin-top: var(--ck-spacing-medium);
+		border-top: 1px solid var(--ck-color-base-border);
+		display: flex;
+		flex-direction: column;
+
+		& .ck-link__button {
+			padding: var(--ck-spacing-medium) var(--ck-spacing-large);
+			border-radius: 0;
+	
+			&>.ck-button__label {
```
Comment (pszczesniak) [inline packages/ckeditor5-link/theme/linkform.css:80]:
```suggestion
			& > .ck-button__label {
```

**Review by pszczesniak — CHANGES_REQUESTED** [review]
Maybe better would be to replace as much as we can the `*` in CSS selectors, it should speed up the matching.

File: packages/ckeditor5-link/src/ui/linkformview.ts [hunk 13]
```diff
@@ -242,49 +282,146 @@ export default class LinkFormView extends View {
...
+	private _createSaveButton(): ButtonView {
+		const t = this.locale!.t;
+		const saveButton = new ButtonView( this.locale );
+
+		saveButton.set( {
+			label: t( 'Insert' ),
+			tooltip: true,
+			withText: true,
+			type: 'submit',
+			class: 'ck-button-insert ck-button-action'
```
Comment (pszczesniak) [inline packages/ckeditor5-link/src/ui/linkformview.ts:330]:
```suggestion
			class: 'ck-button-insert ck-button-action ck-button-bold'
```

**Review by pszczesniak — APPROVED** [review]
